c++/src/H5ArrayType.h

```diff
@@ -61,7 +61,7 @@ class H5_DLLCPP ArrayType : public DataType {
     ArrayType(const hid_t existing_id);
 
     // Noop destructor
-    virtual ~ArrayType() override;
+    virtual ~ArrayType() override = default;
```

Comment:
Benign, but if you're using `override`, `virtual` is redundant.

Reply:
Yes, but it was already like that before my change. This PR is really just trying to fix warnings, especially in header files.

Reply:
Searching for `virtual.*override` gives 80 hits. A cleanup of that redundancy can be for another patch.